external_search.py

```diff
@@ -458,58 +462,92 @@ def query_works(self, query_string, filter=None, pagination=None,
             fields known to be used by the feed generation code.  This
             all comes at a slight performance cost.
         :return: A list of Hit objects containing information about
-            the search results, including the values of any script fields
-            calculated by ElasticSearch during the search process.
+            the search results. This will include the values of any
+            script fields calculated by ElasticSearch during the
+            search process.
         """
-        if not self.works_alias:
+        if isinstance(filter, Filter) and filter.match_nothing is True:
+            # We already know this search should match nothing.  We
+            # don't even need to perform the search.
             return []
 
-        if not pagination:
-            pagination = Pagination.default()
+        pagination = pagination or Pagination.default()
+        query_data = (query_string, filter, pagination)
+        [result] = list(self.query_works_multi([query_data], debug))
```

Comment:
Does calling `list` on the method iterate through the generator to get all the values?

Reply:
Yeah, without the list()... we'd probably be okay here because `[result]` implicitly iterates over the generator.